integration-tests/api/__tests__/admin/product.js

```diff
@@ -1104,6 +1120,128 @@ medusaIntegrationTestRunner({
+          const priceList = await breaking(
+            async () => {
+              return await simplePriceListFactory(dbConnection, {
+                prices: [
+                  {
+                    variant_id: "test-variant",
+                    amount: 100,
+                    currency_code: "usd",
+                  },
+                ],
+              })
+            },
+            async () => {
+              const variantId = v2Product.variants[0].id
+
+              await pricingService.createRuleTypes([
+                {
+                  name: "Region ID",
+                  rule_attribute: "region_id",
+                },
+              ])
+
+              const priceSet = await createVariantPriceSet({
+                container,
+                variantId,
+              })
+
+              const [priceList] = await pricingService.createPriceLists([
+                {
+                  title: "Test price list",
+                  description: "Test",
+                  status: PriceListStatus.ACTIVE,
+                  type: PriceListType.OVERRIDE,
+                  prices: [
+                    {
+                      amount: 5000,
+                      currency_code: "usd",
+                      price_set_id: priceSet.id,
+                      rules: {
+                        region_id: "test-region",
+                      },
+                    },
+                  ],
+                },
+              ])
+
+              return priceList
+            }
+          )
+
+          const res = await api.get(
+            `/admin/products?price_list_id[]=${priceList.id}`,
+            adminHeaders
+          )
+
+          expect(res.status).toEqual(200)
+          expect(res.data.products.length).toEqual(1)
+          expect(res.data.products).toEqual([
+            expect.objectContaining({
+              id: breaking(
+                () => "test-product",
+                () => v2Product.id
+              ),
+              status: "draft",
+            }),
+          ])
+        })
+
+        it("should return products filtered by sales_channel_id", async () => {
+          const { salesChannel, product } = await breaking(
+            async () => {
+              const product = await simpleProductFactory(dbConnection, {
+                id: "product_1",
+                title: "test title",
+              })
+
+              const salesChannel = await simpleSalesChannelFactory(
+                dbConnection,
+                {
+                  name: "test name",
+                  description: "test description",
+                  products: [product],
+                }
+              )
+
+              return { salesChannel, product }
+            },
+            async () => {
+              const salesChannel = await scService.create({
+                name: "Test channel",
+                description: "Lorem Ipsum",
+              })
+
+              await remoteLink.create({
```

Comment:
It's fine to keep this for now since we are still working on the product module, but ideally we try to achieve everything in the API tests through the public interfaces. For example, you'd do an update on the `product` to set the sales channel, rather than creating the link within the test.